utils/manifest-utils/src/policy.rs

```diff
@@ -90,119 +92,83 @@ impl FromStr for Match {
     }
 }
 
-#[derive(Clone, Default)]
+#[derive(Clone)]
 pub struct Keystore {
-    inner: Arc<RwLock<BTreeMap<Box<[u8]>, KeyMeta>>>,
-}
-
-#[derive(Clone, Debug)]
-pub struct KeyMeta {
-    pub scheme: String,
-    pub name: String,
+    inner: Arc<RwLock<X509Store>>,
 }
 
-impl Default for KeyMeta {
+impl Default for Keystore {
     fn default() -> Self {
-        KeyMeta::new(None, None)
-    }
-}
-
-impl KeyMeta {
-    pub fn new(scheme: Option<String>, name: Option<String>) -> Self {
-        KeyMeta {
-            scheme: scheme.unwrap_or_else(|| SCHEME_SECP256K1.to_string()),
-            name: name.unwrap_or_else(|| petname::petname(3, "-")),
+        let store = X509StoreBuilder::new().expect("SSL works").build();
+        Self {
+            inner: Arc::new(RwLock::new(store)),
         }
     }
 }
 
 impl Keystore {
-    pub fn load(path: impl AsRef<Path>) -> anyhow::Result<Self> {
-        let path = path.as_ref();
-        let contents = std::fs::read_to_string(path)
-            .map_err(|e| anyhow::anyhow!("cannot read the keystore file: {}", e))?;
-
-        let map = contents
-            .lines()
-            .map(|s| s.trim())
-            .filter(|s| !s.is_empty() && !s.starts_with('#'))
-            .map(parse_key_entry)
-            .collect::<Result<_, _>>()?;
-
+    /// Reads PEM certificates (or certificate stacks) from `cert_dir` and creates new `X509Store`.
+    pub fn load(cert_dir: impl AsRef<Path>) -> anyhow::Result<Self> {
+        let mut store = X509StoreBuilder::new()?;
+        let cert_dir = std::fs::read_dir(cert_dir)?;
+        for dir_entry in cert_dir {
+            let cert_file = dir_entry?;
+            let cert_file = cert_file.path();
+            let mut cert_file = File::open(cert_file)?;
+            let mut cert_buffer = Vec::new();
+            cert_file.read_to_end(&mut cert_buffer)?;
+            for cert in X509::stack_from_pem(&cert_buffer)? {
```

Comment:
Could support both DER and PEM (& PEM stacks)

Reply:
Alternatively consider adding lookup
https://docs.rs/openssl/latest/openssl/x509/store/struct.X509StoreBuilder.html#method.add_lookup
Precisely this
https://docs.rs/openssl/latest/openssl/x509/store/struct.X509Lookup.html#method.add_dir
(I failed to figure out how to create X509Lookup so I could call #add_dir)

Reply:
DER & PEM support done.